### PR TITLE
Add local node to DB; add middleware that ensures local node exists

### PR DESCRIPTION
**Pull request by adamjford**

### CODE REVIEW

**Comment by adamjford** [comment]
Oops, forgot something here, one sec.

**Comment by adamjford** [comment]
One thing to note is that, because the Author's node field is no longer optional, everyone will need to blow away their databases and start from scratch. I'll repeat this warning in chat once this gets merged.

File: common/middleware/local_node.py
Comment (davidyee) [inline common/middleware/local_node.py:14]: This solution is acceptable within the context of this project but would see DB portability issues if sharing or migrating a server DB from one hostname address to another.

**Review by davidyee — APPROVED** [review]

**Comment by adamjford** [comment]
@davidyee So I attempted to update this such that, upon first request, if it noticed that the local node no longer has the latest host URLs, it updates them. However, for some reason, despite ensuring the code is ran via walking through it with the debugger, it doesn't actually save to the database. Can you see if there's anything obvious I'm missing here?

File: common/middleware/local_node.py
Comment (davidyee) [inline common/middleware/local_node.py:27]: This line will add `host` to the node when you actually meant to update `website_url`.

**Comment by adamjford** [comment]
I'm going to go ahead and merge this, and fix that issue at a later date. Need this functionality to keep going.

File: common/middleware/local_node.py
Comment (davidyee) [inline common/middleware/local_node.py:28]: This line will add `service` to the node when you actually meant to update `service_url`.

Comment (adamjford) [reply common/middleware/local_node.py:28]: UGH. Thank you! Of course, I just hit the dang merge button.

Comment (davidyee) [reply common/middleware/local_node.py:28]: Haha and I was thinking about suggesting that you name `website_url` to `host` when you submitted your initial pull request for the `Node` model. 😹

Comment (adamjford) [reply common/middleware/local_node.py:28]: I'm going to go ahead and fix that right now.

**Pull request merged**
